lib/BloqadeLattices/test/runtests.jl

```diff
@@ -52,6 +52,94 @@ end
+        @test issetequal(generate_sites_in_region(lattice, region).atoms, expected_sites)
+        ## Negative bounds
+        region = Parallelepiped(-4.0)
+        expected_sites = [(0.0,), (-1.0,), (-2.0,), (-3.0,)]
+        println(generate_sites_in_region(lattice, region).atoms)
+        println(expected_sites)
+        @test issetequal(generate_sites_in_region(lattice, region).atoms, expected_sites)
+
+        # Square Lattice
+        lattice = SquareLattice()
+        bounds = zeros((2,2))
+        bounds[1,:] .= (0.0, 2.0)
+        bounds[2,:] .= (2.0, 0.0)
+        region = Parallelepiped(bounds)
+        expected_sites = [(0.0, 0.0), (1.0, 0.0), (0.0, 1.0), (1.0, 1.0)]
+        @test issetequal(generate_sites_in_region(lattice,region).atoms, expected_sites)
+        ## Negative bounds
+        bounds[1,:] .= (2.0, 2.0)
+        bounds[2,:] .= (2.0, -2.0)
+        expected_sites = [(0,0), (1,1), (1,0), (1,-1), (2,1), (2,0), (2,-1), (3,0)]
+        region = Parallelepiped(bounds)
+        @test issetequal(generate_sites_in_region(lattice, region).atoms, expected_sites)
+        
+        # Triangular Lattice
+        lattice = TriangularLattice()
+        bounds = zeros((2,2))
+        bounds[1,:] .= (0, 2)
+        bounds[2,:] .= (2, 0)
+        region = Parallelepiped(bounds)
+        expected_sites = [(0.0, 1.7320508075688772), 
+                          (0.0, 0.0), 
+                          (0.5, 0.8660254037844386), 
+                          (1.0, 1.7320508075688772), 
+                          (1.0, 0.0), 
+                          (1.5, 0.8660254037844386)]
+        @test issetequal(generate_sites_in_region(lattice, region).atoms, expected_sites)
+
+        # Honeycomb Lattice
+        lattice = HoneycombLattice()
+        bounds = zeros((2,2))
+        # create region that slices through lattice unit cells.
+        # Start off with vectors for Honeycomb lattice, 
+        # then rotate and scale as needed
+        a1 = [1.0, 0.0]
+        a2 = [0.5, 0.5 * sqrt(3)]
+        
+        bounds[1,:] .= rot_mat(deg2rad(50)) * a1 * 2
+        bounds[2,:] .= rot_mat(deg2rad(50)) * a2 * 2
+        println(bounds)
+        region = Parallelepiped(bounds)
+        #=
+        expected_sites = [
+            (0.0, 0.0), 
+            (0.5, 0.8660254037844386), 
+            (1.5, 0.8660254037844386), 
+            (0.5, 0.2886751345948129), 
+            (1.0, 1.1547005383792515)
+        ]
+        =#
+        for site in generate_sites_in_region(lattice, region).atoms
+            println(site)
+            println(site ∈ region)
+        end
+        # @test issetequal(generate_sites_in_region(lattice, region).atoms, expected_sites)
+
+    end
+end
+
+@testset "in_region_sanity_check" begin
+    rot_mat(θ) = [cos(θ) -sin(θ); sin(θ) cos(θ)]
+
+    a1 = [1.0, 0.0]
+    a2 = [0.5, 0.5 * sqrt(3)]
+    bounds = zeros((2,2))
+    bounds[1,:] .= rot_mat(deg2rad(50)) * a1 * 2
+    bounds[2,:] .= rot_mat(deg2rad(50)) * a2 * 2
```

Comment:
> 1. \in for regions seems to include points that are definitely not part of the region (see the `in_region_sanity_check` test) and not due to floating point error either

Ah I found the issue, you put the vectors into the rows of the matrix not the rows
```suggestion
    bounds[:,1] .= rot_mat(deg2rad(50)) * a1 * 2
    bounds[:,2] .= rot_mat(deg2rad(50)) * a2 * 2
```